shell/src/main/java/alluxio/cli/validation/SecureHdfsValidationTask.java

```diff
@@ -104,15 +109,20 @@ private boolean validatePrincipalLogin() {
     String realm = matchPrincipal.group("realm");
 
     // Login with principal and keytab
-    String keytab = ServerConfiguration.get(mKeytabProperty);
-    int exitVal =
-        Utils.getResultFromProcess(new String[] {"kinit", "-kt", keytab, principal}).getExitValue();
-    if (exitVal != 0) {
-      System.err.format("Kerberos login failed for %s with keytab %s with exit value %d.%n",
-          principal, keytab, exitVal);
-      System.err.format("Primary is %s, instance is %s and realm is %s.%n",
-          primary, instance, realm);
-      return false;
+    String keytab = conf.get(mKeytabProperty);
+    CommandReturn cr;
+    String[] command = new String[] {"kinit", "-kt", keytab, principal};
+    try {
+      cr = ShellUtils.execCommandWithOutput(command);
+      if (cr.getExitCode() != 0) {
+        System.err.format("Kerberos login failed for %s with keytab %s with exit value %d.%n",
+                principal, keytab, cr.getExitCode());
+        System.err.format("Primary is %s, instance is %s and realm is %s.%n",
+                primary, instance, realm);
+        return false;
+      }
+    } catch (IOException e) {
+      System.err.format("Failed to execute %s with exception: %s%n", Arrays.toString(command), e);
```

Comment:
will this add spaces between the command? or does it print something like `{"arg1", "arg2", ...}`

Reply:
It does the latter.